crates/harp/src/size.rs

```diff
@@ -27,28 +27,28 @@ use crate::r_typeof;
 // `utils::object.size()` is too slow on large datasets and this code path is used trough the
 // variables pane which required more performance.
 // See for more info.
-pub fn r_size(x: SEXP) -> usize {
+pub fn r_size(x: SEXP) -> harp::Result<usize> {
     let mut seen: HashSet<SEXP> = HashSet::new();
 
     let sizeof_node: f64 = harp::parse_eval0(
         "as.vector(utils::object.size(quote(expr = )))",
         R_ENVS.global,
     )
-    .and_then(|x| x.try_into())
-    .unwrap_or(0.);
+    .and_then(|x| x.try_into())?;
 
     let sizeof_vector: f64 = harp::parse_eval_global("as.vector(utils::object.size(logical()))")
-        .and_then(|x| x.try_into())
-        .unwrap_or(0.);
-
-    obj_size_tree(
-        x,
-        R_ENVS.global,
-        sizeof_node as usize,
-        sizeof_vector as usize,
-        &mut seen,
-        0,
-    )
+        .and_then(|x| x.try_into())?;
+
+    harp::try_catch(|| {
```

Comment:
I think this needs a comment about the try-catch, something about the fact that `obj_size_tree()` has the potential to violate some assumptions about R internals